lib/collection.js

```diff
@@ -443,7 +444,7 @@ Collection.prototype.find = deprecateOptions(
     newOptions.db = this.s.db;
 
     // Add the promise library
-    newOptions.promiseLibrary = this.s.promiseLibrary;
+    newOptions.promiseLibrary = PromiseProvider.get(options, this);
```

Comment:
This is within `collection.find` it's not clear to me which methods you can pass in `promiseLibrary` and which you can't. Should this be `PromiseProvider.get(this);` or `PromiseProvider.get(options, this);`? Thoughts?